### PR TITLE
Pipeline, Sofia Nurtdinova - 22FPL2

### PR DESCRIPTION
**Pull request by sofianurtdinova**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1 +1,8 @@
-
+beautifulsoup4==4.12.3
+lxml==5.2.1
+networkx==3.2.1
+requests==2.31.0
+spacy-conll==3.4.0
+spacy-udpipe==1.0.0
+spacy==3.7.4
+stanza==1.8.2
```
Comment (demid5111) [inline requirements.txt:8]: newline

File: lab_6_pipeline/pipeline.py [hunk 2]
```diff
@@ -197,7 +275,7 @@ class PatternSearchPipeline(PipelineProtocol):
     """
 
     def __init__(
-        self, corpus_manager: CorpusManager, analyzer: LibraryWrapper, pos: tuple[str, ...]
+            self, corpus_manager: CorpusManager, analyzer: LibraryWrapper, pos: tuple[str, ...]
```
Comment (demid5111) [inline lab_6_pipeline/pipeline.py:278]: discard changes in function definition

File: lab_6_pipeline/pipeline.py [hunk 3]
```diff
@@ -99,6 +163,8 @@ def analyze(self, texts: list[str]) -> list[StanzaDocument | str]:
         Returns:
             list[StanzaDocument | str]: List of documents
         """
+        conllu_annotations = [self._analyzer(text)._.conll_str for text in texts]
```
Comment (demid5111) [inline lab_6_pipeline/pipeline.py:166]: unneeded variable, return immediately

File: lab_6_pipeline/pipeline.py [hunk 4]
```diff
@@ -62,11 +108,17 @@ def __init__(
...
+        self._corpus = corpus_manager
+        self.analyzer = analyzer
 
     def run(self) -> None:
         """
         Perform basic preprocessing and write processed text to files.
         """
+        for article in self._corpus.get_articles().values():
+            to_cleaned(article)
+            article.set_conllu_info(self.analyzer.analyze(split_by_sentence(article.text)))
```
Comment (demid5111) [inline lab_6_pipeline/pipeline.py:120]: no. we have to pass to analyzer a batch of documents and not batch of sentences. this idea should considerably change the logic of this method

File: lab_6_pipeline/pipeline.py [hunk 5]
```diff
@@ -53,7 +99,7 @@ class TextProcessingPipeline(PipelineProtocol):
     """
 
     def __init__(
-        self, corpus_manager: CorpusManager, analyzer: LibraryWrapper | None = None
+            self, corpus_manager: CorpusManager, analyzer: LibraryWrapper | None = None
```
Comment (demid5111) [inline lab_6_pipeline/pipeline.py:102]: .

File: lab_6_pipeline/pipeline.py [hunk 6]
```diff
@@ -27,16 +48,40 @@ def __init__(self, path_to_raw_txt_data: pathlib.Path) -> None:
...
+                    raw.stat().st_size == 0 or meta.stat().st_size == 0:
+                raise InconsistentDatasetError
 
     def _scan_dataset(self) -> None:
         """
         Register each dataset entry.
         """
+        for file in list(self.path_to_raw_txt_data.glob("*_raw.txt")):
+            article_id = get_article_id_from_filepath(file)
+            self._storage[article_id] = from_raw(file, Article(url=None, article_id=article_id))
```
Comment (demid5111) [inline lab_6_pipeline/pipeline.py:84]: .

**Review by demid5111 — CHANGES_REQUESTED** [review]
CI is also red. keep going!

File: lab_6_pipeline/pipeline.py (hunk 2 above)
Comment (sofianurtdinova) [reply lab_6_pipeline/pipeline.py:278]: reverted

File: lab_6_pipeline/pipeline.py (hunk 3 above)
Comment (sofianurtdinova) [reply lab_6_pipeline/pipeline.py:166]: fixed

File: lab_6_pipeline/pipeline.py (hunk 5 above)
Comment (sofianurtdinova) [reply lab_6_pipeline/pipeline.py:102]: changed

File: lab_6_pipeline/pipeline.py (hunk 6 above)
Comment (sofianurtdinova) [reply lab_6_pipeline/pipeline.py:84]: hope it is anticipated that it's the comprehension here which is needed...

File: requirements.txt (hunk 1 above)
Comment (sofianurtdinova) [reply requirements.txt:8]: done...

File: lab_6_pipeline/pipeline.py [hunk 7]
```diff
@@ -62,11 +111,21 @@ def __init__(
...
+        self.analyzer = analyzer
 
     def run(self) -> None:
         """
         Perform basic preprocessing and write processed text to files.
         """
+        docs = [article.text for article in self._corpus.get_articles().values()]
+        if self.analyzer:
+            docs = self.analyzer.analyze(docs)
+        for i, article in enumerate(self._corpus.get_articles().values()):
```
Comment (demid5111) [inline lab_6_pipeline/pipeline.py:124]: zip with docs and iterate through all values simultaneously

File: lab_6_pipeline/pipeline.py [hunk 8]
```diff
@@ -27,16 +49,42 @@ def __init__(self, path_to_raw_txt_data: pathlib.Path) -> None:
...
+                    id_num + 1 != get_article_id_from_filepath(raw) or \
+                    raw.stat().st_size == 0 or meta.stat().st_size == 0:
+                raise InconsistentDatasetError
 
     def _scan_dataset(self) -> None:
         """
         Register each dataset entry.
         """
+        self._storage = {
+            get_article_id_from_filepath(file):
```
Comment (demid5111) [inline lab_6_pipeline/pipeline.py:84]: very good. 🏅

**Review by demid5111 — CHANGES_REQUESTED** [review]
it is ok. CI is red and mark is only 8

File: lab_6_pipeline/pipeline.py (hunk 7 above)
Comment (sofianurtdinova) [reply lab_6_pipeline/pipeline.py:124]: I'm not sure that I have got you right, like this? If not, I'll rework

Comment (sofianurtdinova) [reply lab_6_pipeline/pipeline.py:123]: oh, no I think I got the idea about using zip...

File: lab_6_pipeline/pipeline.py (hunk 5 above)
Comment (demid5111) [reply lab_6_pipeline/pipeline.py:108]: no

File: lab_6_pipeline/pipeline.py (hunk 2 above)
Comment (demid5111) [reply lab_6_pipeline/pipeline.py:312]: no

File: lab_6_pipeline/pipeline.py [hunk 9]
```diff
@@ -253,6 +389,19 @@ def main() -> None:
...
+    pipeline.run()
+    stanza_analyzer = StanzaAnalyzer()
+    pipeline = TextProcessingPipeline(corpus_manager, stanza_analyzer)
+    pipeline.run()
+    visualizer_pos = POSFrequencyPipeline(corpus_manager, stanza_analyzer)
+    visualizer_pos.run()
+    # visualizer_pat = PatternSearchPipeline(corpus_manager,
+    #                                        stanza_analyzer,
+    #                                        ("VERB", "NOUN", "ADP"))
+    # visualizer_pat.run()
```
Comment (demid5111) [inline lab_6_pipeline/pipeline.py:404]: before defense, if not ready, remove all commented code.

File: lab_6_pipeline/pipeline.py [hunk 10]
```diff
@@ -242,7 +371,14 @@ def _find_pattern(self, doc_graphs: list) -> dict[int, list[TreeNode]]:
...
         """
-
+        # patterns = {}
+        # for sentence_id, graph in enumerate(doc_graphs):
+        #     ideal_graph = DiGraph()
+        #     ideal_graph.add_nodes_from((i, {'label': label})
+        #                                for i, label in enumerate(self._node_labels))
+        #     ideal_graph.add_edges_from((i, i + 1)
+        #                                for i in range(len(self._node_labels) - 1))
+        #     matcher = GraphMatcher(ideal_graph, graph)
```
Comment (demid5111) [inline lab_6_pipeline/pipeline.py:381]: .

**Review by demid5111 — CHANGES_REQUESTED** [review]
I see that you are trying to make for 10. there is still some time, so keep trying. however, I would recommend first securing 8, therefore make CI green first

**Pull request closed without merging**